python_modules/dagster/dagster/_core/definitions/events.py

```diff
@@ -443,17 +453,32 @@ def __new__(
             check.tuple_param(asset_key, "asset_key", of_type=str)
             asset_key = AssetKey(asset_key)
 
+        check.opt_mapping_param(tags, "tags", key_type=str, value_type=str)
+        if any([not tag.startswith(SYSTEM_TAG_PREFIX) for tag in tags or {}]):
+            check.failed(
+                "Users should not pass values into the tags argument for AssetMaterializations. The tags argument is reserved for system-populated tags."
+            )
+
         metadata = check.opt_mapping_param(metadata, "metadata", key_type=str)
         metadata_entries = check.opt_sequence_param(
             metadata_entries, "metadata_entries", of_type=(MetadataEntry, PartitionMetadataEntry)
         )
 
+        serializable_partition: Optional[Union[str, MultiDimensionalPartitionKey]] = None
```

Comment:
The name `serializable_partition` tripped me up a tiny bit here, because it implies that the `partition` argument supplied by the user isn't serializable. But both are serializable, right?